### DISCUSSION TITLE
HASPmota and slider range

### DISCUSSION THREAD

**Discussion by waltkind** [arendst/Tasmota#20581] [Q&A] [answered]
I am trying to implement a slider to a screen with values between 0 and 255.
As there are no special attributes for slider in the HASPmota documentation I looked at the openHASP documentation and took the sample:
```berry
{"page":1,"id":6,"obj":"slider","x":20,"y":250,"w":200,"h":20,"min":15,"max":35}

``` 
But no luck. The values vary between 0 and 100. 
Any alteration of 'min' or 'max' has no effect.
What is the right syntax for a range?

**Comment by s-hadinger** [comment]
Ah too bad. The undernlying LVGL object does not have `set_min` nor `set_max` but only `set_range`. So I will need to add the helper functions to implement them. This should be quick

**Comment by s-hadinger** [comment, marked as answer]
It's fixed now in the latest version

**Reply by waltkind** [reply]
Thank you for fixing the problem so fast.
Now I am struggeling with the new build. Because I have a Lanbon L8 I need a special build with esp32-fix for the PSRAM.
When I compile a new build on Gitpod with changing to the development branch in the bottom line, I geta new build that shows on my device as 13.3.0.2 and the slider is not working right.
On github I see that the development branch is at 13.3.0.4.
How can I  get gitpod to use this branch?

Got it. I had to Synchronize Changes and now it is 13.3.0.4 and the slider works. Thanks.

**Comment by waltkind** [comment]
I am sorry to say, that now I see other problems. The values of the slider are OK now, but I can not set any attributes. If I set bg_color, bg_color10, bg_color20 or radius20 the screen stays clean. Nothing is shown.

**Reply by s-hadinger** [reply]
It does work for me:

```berry
p1b6.bg_color = "#FF0000"
p1b6.bg_opa = 255
p1b6.bg_color10 = "#FFFFFF"
p1b6.bg_color20 = "#0000FF"
p1b6.radius = 0
p1b6.radius20 = 1
```

Or in the `pages.jsonl`:
```json
{"page":1,"comment":"---------- Page 1 ----------"}
{"id":0,"bg_color":"#0000A0","text_color":"#FFFFFF"}
{"id":6,"obj":"slider","x":20,"y":200,"w":200,"h":20,"min":15,"max":35,"val":20,"bg_color":"#FF0000","bg_opa":255,"bg_color10":"#FFFFFF","bg_color20":"#0000FF","radius":0,"radius20":1}
```

**Reply by s-hadinger** [reply]
One possible reason can be that the json is invalid, I just realized that invalid JSON lines are silently ignored

**Reply by sfromis** [reply]
You can validate single entries using https://jsonlint.com/
(Multiple lines from JSONL are not valid JSON, unless taken one by one)

**Reply by waltkind** [reply]
Thanks to your comments I rechecked all and really there had been a typo in the jsonl-line. So nothing was to be seen.
